daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -2829,13 +2829,19 @@ sealed abstract class LengthExprBase(
   ) {
 
   protected final def leafReferencedElements = {
-    val arg = args(0).asInstanceOf[PathExpression]
+    val arg = args(0) match {
+      case arg: PathExpression => arg
+      case _ => Assert.invariantFailed("NodeInfo.Exists, but arg was not a PathExpression.")
```

Comment:
It's annoying, but invariantFailed always wants the COVERAGE-ON/COVERAGE-OFF stuff around it. 
I wish we could somehow build that in so it is implied for these assertions, but I don't know how. 

Maybe there's a way to configure coverty so that it has a file that declares these sort of things so that one does not need to surround them?

Reply:
> It's annoying, but invariantFailed always wants the COVERAGE-ON/COVERAGE-OFF stuff around it. I wish we could somehow build that in so it is implied for these assertions, but I don't know how.

I bracketed the invariant with COVERAGE-ON/COVERAGE-OFF. I checked the docs but there doesn't seem to be a way to configure this at function invocation time.